### PR TITLE
Enhancement of Jac reference

### PR DESCRIPTION
**Pull request by Thamirawaran**
## **Description**

This PR includes two key updates:

1. Jac Reference Examples Update:
       -  Added the disengage command in visit_statements.jac to prevent overflow.
       -  Corrected a typo in match_patterns.jac (flaot -> float).
       - Implemented a bug fix for enum functionality to have the ability inside an enum.
       -  Bug fix: Updated normalize function for enums to support unparse
       - Updated examples for enums, imports, match patterns, names, and references.
      
2. Initial Version of Data Spatial Examples:
     - Added an FAQ section with Data Spatial Examples for the Data Spatial documentation.

### CODE REVIEW

File: jac/examples/data_spatial/custom_edge.jac
Comment (Jayanaka-98) [inline jac/examples/data_spatial/custom_edge.jac:1]: Show how we can filter by edge attributes.

File: jac/examples/data_spatial/define_walker.jac [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+edge a{
+}
+
+walker W{
+    can create with `root entry;
+}
+
+:walker:W:can:create{
+    Start = A(5);
+    root +:a:+> Start;
```
Comment (Jayanaka-98) [inline jac/examples/data_spatial/define_walker.jac]: Huse 'here' instead of root

File: jac/examples/data_spatial/filtering.jac
Comment (Jayanaka-98) [inline jac/examples/data_spatial/filtering.jac:1]: filtering by node attributes should be also included

File: jac/support/jac-lang.org/mkdocs.yml
Comment (Jayanaka-98) [inline jac/support/jac-lang.org/mkdocs.yml:1]: Remove the reference.md if form yml if it is empty. After completing that add it later in a seperetare PR

**Review by Jayanaka-98 — CHANGES_REQUESTED** [review]

**Review by kugesan1105 — APPROVED** [review]
This is good to merge.

File: jac/support/jac-lang.org/mkdocs.yml
Comment (kugesan1105) [reply jac/support/jac-lang.org/mkdocs.yml:1]: done!

**Review by marsninja — APPROVED** [review]

**Comment by marsninja** [comment]
Nice work @Thamirawaran !!!

**Pull request merged**